Microsoft.Toolkit.HighPerformance/Streams/IBufferWriterStream{TWriter}.Memory.cs

```diff
@@ -0,0 +1,76 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+#if SPAN_RUNTIME_SUPPORT
```

Comment:
Remind me, does this mean it only works on .NET Standard 2.1 or they have to have the Span package for .NET Standard 2?

Reply:
That constant specifically refers to runtime support for `Span<T>`, as in, when the target assumes that the runtime can support the "fast `Span<T>`" version and comes bundled with `Span<T>` out of the box, meaning that we have access to `Span<T>`-based overloads for all the various APIs in the BCL, and not just as an external type that added after the fact (like is the case on .NET Standard 2.0). In this case here we're overriding some `Span<T>`-based `Stream` APIs, which are only available in that case, ie. on either .NET Standard 2.1 and up, or on .NET Core 2.1 (which doesn't support .NET Standard 2.1 but still exposes the same APIs in this area). This custom constant is basically the union of both targets specifically with respect to `Span<T>` APIs, so we can offer the best support for all these various runtimes.

I added a full explanation for that compiler constant right in the `.csproj` file too for future reference 🙂

https://github.com/windows-toolkit/WindowsCommunityToolkit/blob/846afb25bf3d6fb85b78bf810d060b66dcf81a1b/Microsoft.Toolkit.HighPerformance/Microsoft.Toolkit.HighPerformance.csproj#L68-L75